### DISCUSSION TITLE
How to configure the runtime path (/run/firejail) to a different directory?

### DISCUSSION THREAD

**Discussion by bodachaitanya** [netblue30/firejail#5258] [Q&A] [answered]
Hi Firejail team,

When firejail is run, it creates /run/firejail directory on the node.
I was looking for an option to change to different directory path other than /run/firejail.
I have tried to create softlink to some other directory path as well (eg: /run -> /runtime/run), but it didn't work and getting error when firejail is run.
Could anyone help in providing me suitable option/configuration to change the directory path from /run/firejail to some other path.

Thanks & Regards,
Chaitanya

**Comment by rusty-snake** [comment]
https://github.com/netblue30/firejail/blob/fb87e9a4d911a2b6982c6d64bcc238c72aa89d89/src/include/rundefs.h#L24-L25

**Comment by rusty-snake** [comment, marked as answer]
XY Problem, you didn't say why you need/want this but maybe a bind-mount solves your problem.

**Reply by bodachaitanya** [reply]
/run is not a standard partition we use. bind-mount is definitely a way, but having a soft link supported to different path (say /run -> /data/runtime/run) would be more useful.

**Reply by poly2it** [reply]
This is not an X-Y problem. The runtime directory should not be bound to a static out-of-home path. This for example breaks running firejail without a full installation on NixOS, as normal users cannot normally modify /run there.

**Reply by poly2it** [reply]
@rusty-snake

**Reply by rusty-snake** [reply]
Can normal users install a suid root binary in NixOS?

**Reply by rusty-snake** [reply]
> This is not an X-Y problem

How w/o why is XY because even if the how is correct you can not know that it is without knowing the why.

**Reply by rusty-snake** [reply]
> The runtime directory should not be bound to a static *out-of-home* path.

I'm very concerned that this would cause more harm than good for firejail.

Also general, runtime files do not belong in `$HOME`. `$XDG_RUNTIMR_DIR` is for them.

**Reply by kmk3** [reply]
> I'm very concerned that this would cause more harm than good for firejail.

I'd think so as well.

The following answer probably also applies here:

* https://github.com/netblue30/firejail/discussions/6554#discussioncomment-11381785
